sources/HUBViewControllerImplementation.m

```
@@ -1314,6 +1337,103 @@ - (void)removeComponentWrapperFromLookupTables:(nullable HUBComponentWrapper *)c
     }
 }
 
+- (void)scrollToRootComponentAtIndex:(NSUInteger)componentIndex
+                      scrollPosition:(UICollectionViewScrollPosition)scrollPosition
+                            animated:(BOOL)animated
+                          completion:(void (^)())completionHandler
+{
+    NSUInteger const componentCount = (NSUInteger)[self.collectionView numberOfItemsInSection:0];
+    if (componentIndex == NSNotFound || componentIndex >= componentCount) {
```

@JohnSundell – do we normally assert these things or fail silently?